sljit_src/sljitNativeMIPS_32.c

```diff
@@ -30,17 +30,37 @@ static sljit_s32 emit_copysign(struct sljit_compiler *compiler, sljit_s32 op,
 		sljit_sw src1, sljit_sw src2, sljit_sw dst)
 {
 	int is_32 = (op & SLJIT_32);
-	sljit_ins fr_pair = is_32 ? 0 : (1 << 11);
+	sljit_ins mfhc = MFC1, mthc = MTC1;
+	sljit_ins src1_r = FS(src1), src2_r = FS(src2), dst_r = FS(dst);
+
+	if (!is_32) {
+#if defined(SLJIT_MIPS_REV) && SLJIT_MIPS_REV >= 2
+		if (cpu_feature_list & CPU_FEATURE_FR) {
+			mfhc = MFHC1;
+			mthc = MTHC1;
+#else
+		if (0) {
```

Comment:
This construct does not looks nice. Maybe we don't need `#if` at all. Or use a `switch` or a `do while(0)` with `break`

Reply:
`#if` is needed at least to protect the use of `MFHC1` and `MTHC1` which are only defined (and available) if SLJIT_MIPS_REV >= 2.

not sure I understood how to use the alternatives you proposed but tried to make it a little less ugly

Reply:
```
switch (cpu_feature_list & CPU_FEATURE_FR) {
#if ...
case CPU_FEATURE_FR:
  break;
#endif
default:
  break;
}